plugins/generator-1.19.4/forge-1.19.4/templates/livingentity/livingentity.java.ftl

```diff
@@ -246,6 +350,22 @@ public class ${name}Entity extends ${extendsClass} <#if data.ranged>implements R
 	}
 	</#if>
 
+	<#if data.isWanderingTraderType()>
+	@Override protected SoundEvent getTradeUpdatedSound(boolean hasContent) {
+		return ForgeRegistries.SOUND_EVENTS.getValue(new ResourceLocation(hasContent ?
+			"${(data.tradeFullUpdateSound?has_content && data.tradeFullUpdateSound.getMappedValue()?has_content)?then(data.tradeFullUpdateSound, "intentionally_empty")}"
+		:
+			"${(data.tradeEmptyUpdateSound?has_content && data.tradeEmptyUpdateSound.getMappedValue()?has_content)?then(data.tradeEmptyUpdateSound, "intentionally_empty")}"
+		));
+	}
+
+	@Override public SoundEvent getNotifyTradeSound() {
+		return ForgeRegistries.SOUND_EVENTS.getValue(new ResourceLocation(

```

Comment:
Maybe we should have a new `toSoundEvent(sound)` function in `mcelements.ftl` because it becomes hard to read long tests like here? Or at least `sounds.yaml` mappings with `intentionally_empty` default entry?

Reply:
You can refactor that on master in separate PR and this PR then adopts this

Reply:
Which option?

Reply:
> Or at least sounds.yaml mappings with intentionally_empty default entry?

I think this is already implemented so maybe it can just be simplified now. But then all other cases can be simplified in the PR I suggested. However make sure it works like that first

Reply:
@Defeatomizer do you want to take that optimization PR or should I do it?

Reply:
If you have more free time, I don't mind it being taken over.

Reply:
I can confirm that in this PR sound events can reduce checks back to direct mapping use as default value takes care of intentionally_empty

Reply:
So no need to check if mapped value is present?

Reply:
Yep

Reply:
Afaik value is always present, but can be empty

Reply:
Or do we set it to null? In this case, helper is still needed